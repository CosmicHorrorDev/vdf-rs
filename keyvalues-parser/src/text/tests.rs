use std::{error::Error, fs, path::Path};

use crate::Vdf;

type BoxedResult<T> = Result<T, Box<dyn Error>>;

fn read_asset_file(file_name: &str) -> BoxedResult<String> {
    let val = fs::read_to_string(Path::new("tests").join("assets").join(file_name))?;
    Ok(val)
}

fn snapshot_test_parse_and_render(file_name: &str) -> BoxedResult<()> {
    let vdf_text = read_asset_file(file_name)?;
    let vdf = Vdf::parse(&vdf_text)?;
    insta::assert_ron_snapshot!(vdf);

    let rendered = vdf.to_string();
    insta::assert_snapshot!(rendered);

    Ok(())
}

macro_rules! parse_render_test {
    ($func_name:ident, $file_name: literal) => {
        #[test]
        fn $func_name() -> BoxedResult<()> {
            snapshot_test_parse_and_render($file_name)
        }
    };
}

parse_render_test!(basic, "basic.vdf");
parse_render_test!(app_manifest, "app_manifest.vdf");
parse_render_test!(comments, "comments.vdf");
parse_render_test!(unquoted_strings, "unquoted_strings.vdf");
parse_render_test!(special_charaters, "special_characters.vdf");
parse_render_test!(app_info, "app_info.vdf");
