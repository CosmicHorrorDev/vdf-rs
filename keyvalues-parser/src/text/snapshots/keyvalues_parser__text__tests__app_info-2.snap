---
source: keyvalues-parser/src/text/tests.rs
expression: rendered

---
"896660"
{
	"branches"
	{
		"experimental"
		{
			"buildid"	"6263839"
			"description"	"Experimental version of Valheim"
			"pwdrequired"	"1"
			"timeupdated"	"1613728251"
		}
		"public"
		{
			"buildid"	"6246034"
			"timeupdated"	"1613558776"
		}
		"unstable"
		{
			"buildid"	"6246034"
			"description"	"Unstable test version of valheim"
			"pwdrequired"	"1"
			"timeupdated"	"1613469743"
		}
	}
	"common"
	{
		"ReleaseState"	"released"
		"associations"
		{
		}
		"clienticon"	"c55a6b50b170ac6ed56cf90521273c30dccb5f12"
		"clienttga"	"35e067b9efc8d03a9f1cdfb087fac4b970a48daf"
		"gameid"	"896660"
		"icon"	"1aab0586723c8578c7990ced7d443568649d0df2"
		"logo"	"233d73a1c963515ee4a9b59507bc093d85a4e2dc"
		"logo_small"	"233d73a1c963515ee4a9b59507bc093d85a4e2dc_thumb"
		"name"	"Valheim Dedicated Server"
		"osarch"	""
		"oslist"	"windows,linux"
		"parent"	"892970"
		"type"	"Tool"
	}
	"config"
	{
		"installdir"	"Valheim dedicated server"
		"launch"
		{
			"0"
			{
				"config"
				{
					"oslist"	"linux"
				}
				"executable"	"start_server_xterm.sh"
				"type"	"server"
			}
			"1"
			{
				"config"
				{
					"oslist"	"windows"
				}
				"executable"	"start_headless_server.bat"
				"type"	"server"
			}
		}
	}
	"depot"
	{
		"config"
		{
			"oslist"	"windows"
		}
		"depotfromapp"	"1007"
		"id"	"1004"
		"manifests"
		{
			"public"	"6473168357831043306"
		}
		"maxsize"	"39546856"
		"name"	"Steamworks SDK Redist (WIN32)"
	}
	"depot"
	{
		"config"
		{
			"oslist"	"macos"
		}
		"depotfromapp"	"1007"
		"id"	"1005"
		"manifests"
		{
			"public"	"2135359612286175146"
		}
		"name"	"Steamworks SDK Redist (OSX32)"
	}
	"depot"
	{
		"config"
		{
			"oslist"	"linux"
		}
		"depotfromapp"	"1007"
		"id"	"1006"
		"manifests"
		{
			"public"	"6688153055340488873"
		}
		"maxsize"	"59862244"
		"name"	"Steamworks SDK Redist (LINUX32)"
	}
	"depot"
	{
		"config"
		{
			"oslist"	"linux"
		}
		"encryptedmanifests"
		{
			"experimental"
			{
				"encrypted_gid_2"	"BEDF872D73873D16C025EF87E27C2BDB"
				"encrypted_size_2"	"2559486959C6E5DCEA5C71ED32BA9080"
			}
		}
		"id"	"896661"
		"manifests"
		{
			"public"	"521795651741005384"
		}
		"maxsize"	"985409357"
		"name"	"Valheim dedicated server Linux"
	}
	"depot"
	{
		"config"
		{
			"oslist"	"windows"
		}
		"encryptedmanifests"
		{
			"experimental"
			{
				"encrypted_gid_2"	"9FD2B7B42FACB1D1FC439DD83ED2BED9"
				"encrypted_size_2"	"B2D602E667364DEDCB7C3D6EE9AA7374"
			}
		}
		"id"	"896662"
		"manifests"
		{
			"public"	"5449924312569304795"
		}
		"maxsize"	"963189471"
		"name"	"Valheim dedicated server Windows"
	}
}

